### ISSUE TITLE
Address 'Code Clone Detection' results

### ISSUE THREAD

**Issue by mrlacey** [Microsoft/WindowsTemplateStudio#933]
The 'Code Clone Detection' functionality in VSEnt highlights some areas where we can improve the generator code base.

I see 705 lines of Cloned Code.

Some of the duplication in test code might not be worth removing but there's strong evidence of someone having abused cut-and-paste when it comes to other code.

**Comment by crutkas** [comment]
Think this may be a good idea to do.

**Comment by sibille** [comment]
ok!

**Comment by sibille** [comment]
eliminated the exact matches, moving to 2.4 for the rest

**Comment by crutkas** [comment]
is this a tracked item for this release?

**Comment by sibille** [comment]
I'd prefer to do this on 3.0 branch

**Comment by crutkas** [comment]
lower priority due to wizard code, lets do in 3.0

Are we doing the test on generated projects as well?

**Comment by mrlacey** [comment]
> Are we doing the test on generated projects as well?

As we aim to only have minimal, necessary, code in the generated projects and the generated code _should_ be easy to understand, I would not expect any clones to be found. If any were found, removing the duplication would likely make the generated code harder to read/understand.

**Comment by dgomezc** [comment]
We have refactored the duplicated code in the UI and Core projects, and some cases that we thought convenient in the tests. We have ignored the rest of the duplicated code in the tests (adding .codeclonesettings files) since they are initializations and we think it is more understandable that way.

In the generated projects, there is only duplicated code when adding Ink Pages, since the initializations of the services are very similar, and in a configuration method of the CameraControl, which contains 2 similar switch statements. We think that the code is easier to understand as it is, so we think there is no action to take here.
